pytorch_lightning/trainer/trainer.py

```diff
@@ -1010,6 +1010,7 @@ def _run_train(self) -> None:
                 self.training_type_plugin.reconciliate_processes(traceback.format_exc())
             # give accelerators a chance to finish
             self.accelerator.on_train_end()
+            self.on_expection()
```

Comment:
As discussed somewhere else, this should not call this and instead the `on_train_end` hook should be called

Reply:
I don't believe there is a `on_train_end` outside of the hook on the Trainer now. Mind sharing more thoughts there ?

Reply:
@tchaton I think fault tolerant should save a special checkpoint, not through the ModelCheckpoint callback (which is anyway not guaranteed to be present) because it would have side effects within that callback state. 

I think it could have a special name so we know this one should be used to resume a faulty experiment.

For these reasons I strongly think that we should just call `trainer.save_checkpoint()` here. 

What are the arguments against?

Reply:
I added an explicit `trainer.save_checkpoint()` call there. 
Have a look and if it's ok I can add a small test.